File: mirai-japt/src/main/java/net/mamoe/mirai/japt/BlockingContacts.java
package net.mamoe.mirai.japt;

import net.mamoe.mirai.Bot;
import net.mamoe.mirai.contact.Group;
import net.mamoe.mirai.contact.Member;
import net.mamoe.mirai.contact.QQ;
import net.mamoe.mirai.japt.internal.BlockingBotImpl;
import net.mamoe.mirai.japt.internal.BlockingGroupImpl;
import net.mamoe.mirai.japt.internal.BlockingMemberImpl;
import net.mamoe.mirai.japt.internal.BlockingQQImpl;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

/**
 * 构造阻塞式的联系人.
 */
public final class BlockingContacts {
    @NotNull
    public static BlockingQQ createBlocking(@NotNull QQ qq) {
        return new BlockingQQImpl(Objects.requireNonNull(qq));
    }

    @NotNull
    public static BlockingGroup createBlocking(@NotNull Group group) {
        return new BlockingGroupImpl(Objects.requireNonNull(group));
    }

    @NotNull
    public static BlockingMember createBlocking(@NotNull Member member) {
        return new BlockingMemberImpl(Objects.requireNonNull(member));
    }

    @NotNull
    public static BlockingBot createBlocking(@NotNull Bot bot) {
        return new BlockingBotImpl(Objects.requireNonNull(bot));
    }
}

File: mirai-japt/src/main/kotlin/net/mamoe/mirai/japt/BlockingContacts.kt
@file:Suppress("NOTHING_TO_INLINE", "unused")

package net.mamoe.mirai.japt

import net.mamoe.mirai.contact.Group
import net.mamoe.mirai.contact.Member
import net.mamoe.mirai.contact.QQ

inline fun Group.blocking(): BlockingGroup = BlockingContacts.createBlocking(this)
inline fun QQ.blocking(): BlockingQQ = BlockingContacts.createBlocking(this)
inline fun Member.blocking(): BlockingMember = BlockingContacts.createBlocking(this)